### ISSUE TITLE
Rename "Heap"?

### ISSUE THREAD

**Issue by sunfishcode** [CraneStation/cranelift#588]
"Heap" is a pretty generic term, but Cranelift's `Heap` is a very specific construct. What if we renamed Cranelift's "Heap" to "LinearMemory", since it's essentially a WebAssembly linear memory with a few more knobs exposed?

And perhaps we could shorten it to "linearmem" in some places, so that `heap_addr` would be `linearmem_addr`. Or maybe `linmem_addr`? Or maybe `linear_addr`?

**Comment by lachlansneff** [comment]
I actually like the term `Heap`. It's short, concise, and once you know what it means in this context, it pretty clear.

**Comment by bnjbvr** [comment]
I certainly understand how it can be confusing: heap can refer to memory allocated by `malloc` for some, and with features that might be introduced to support the wasm GC, we'll also be talking about the GC heap itself. I like the explicitness of calling the current Heap "linear memory" instead and using `linearmem` as a prefix.

**Comment by lachlansneff** [comment]
What about just Memory and `mem` as the prefix>?

**Comment by pchickey** [comment]
I agree that heap is confusing, but lime is a bit too cute. `linearmem` and `linear_addr` sound good to me.
